src/Components/Modal.js

```diff
@@ -3,54 +3,75 @@ import ReactDOM from 'react-dom';
 
 import '../styles/Modal.css'
 
-const Modal = ({
-  title,
-  poster,
-  backdrop,
-  averageRating,
-  releaseDate,
-  overview,
-  genres,
-  budget, 
-  revenue, 
-  tagline,
-  id,
-  selectedMovie,
-  isModalOpen,
-  toggleModal
-}) => {
-  const myStyle = {
-    backgroundImage: `url(${backdrop})`
+import { fetchSingleMovie } from '../api-Calls'
+
+class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: this.props.selectedMovie,
+      title: null,
+      poster: null,
+      backdrop: null,
+      averageRating: null,
+      releaseDate: null,
+      overview: null,
+      genres: null,
+      budget: null,
+      revenue: null,
+      tagline: null,
+      isModalLoading: true,
+    }
   }
 
-  if (isModalOpen && selectedMovie === id) {
-    return ReactDOM.createPortal(
+
+componentDidMount = () => {
+  fetchSingleMovie(this.state.movie)
+  .then((data) => this.setState({
+    title: data.movie.title,
+    poster: data.movie.poster_path,
+    backdrop: data.movie.backdrop_path,
+    averageRating: Number(data.movie.average_rating).toFixed(2),
+    releaseDate: data.movie.release_date,
+    overview: data.movie.overview,
+    runtime: data.movie.runtime,
+    genres: data.movie.genres,
+    budget: data.movie.budget,
+    revenue: data.movie.revenue,
+    tagline: data.movie.tagline,
+    isModalLoading: false
+  }))
+}
+
+render = () => {
+  const myStyle = {
+        backgroundImage: `url(${this.state.backdrop})`
+      }
+
+  return ReactDOM.createPortal(
```

Comment:
This works and portals are encouraged for modals. 

When adding complexity like this, please consider the following: 

- Be sure to be able to back up why you need this modal to specifically outside of the normal DOM hierarchy. It works without this (although the background of the modal isn't darkened...), so why should it be a portal? You might have a great reason - just be ready to back it up. 
- Consider what happens when you need to add in functionality or refactor. Do portals play nicely with the new router? (I don't know the answer to this. It's probably just fine! But it's something you should consider when adding in complexity)

As a side note - I'm against adding HTML to the index.html directly in React or any FE framework (that I've worked in so far) but not all devs agree. The general idea of not doing it is for readability (devs can find everything they need to know about a component in one spot - the component).